--- origsrc/gst-plugins-good-0.10.30/configure.ac	2011-06-15 17:46:51.000000000 -0500
+++ src/gst-plugins-good-0.10.30/configure.ac	2011-12-18 23:43:14.106429700 -0600
@@ -443,6 +443,12 @@ int main ()
   AC_SUBST(HAVE_DIRECTSOUND)
 ])
 
+dnl *** Win32 WaveOut ***
+translit(dnm, m, l) AM_CONDITIONAL(USE_WAVEFORM, true)
+AG_GST_CHECK_FEATURE(WAVEFORM, [Win32 WaveForm], waveformsink, [
+  AC_CHECK_HEADER(mmsystem.h, HAVE_WAVEFORM="yes", HAVE_WAVEFORM="no", [#include <windows.h>])
+])
+
 dnl *** OSS audio *** (Linux, *BSD)
 translit(dnm, m, l) AM_CONDITIONAL(USE_OSS, true)
 AG_GST_CHECK_FEATURE(OSS, [OSS audio], ossaudio, [
@@ -1076,6 +1082,7 @@ AM_CONDITIONAL(USE_SOUP, false)
 AM_CONDITIONAL(USE_SPEEX, false)
 AM_CONDITIONAL(USE_SUNAUDIO, false)
 AM_CONDITIONAL(USE_TAGLIB, false)
+AM_CONDITIONAL(USE_WAVEFORM, false)
 AM_CONDITIONAL(USE_WAVPACK, false)
 AM_CONDITIONAL(USE_X, false)
 AM_CONDITIONAL(USE_XSHM, false)
--- origsrc/gst-plugins-good-0.10.30/sys/Makefile.am	2011-06-13 18:24:12.000000000 -0500
+++ src/gst-plugins-good-0.10.30/sys/Makefile.am	2011-12-18 23:30:48.465781500 -0600
@@ -23,6 +23,12 @@ else
 DIRECTSOUND_DIR=
 endif
 
+if USE_WAVEFORM
+WAVEFORM_DIR=waveform
+else
+WAVEFORM_DIR=
+endif
+
 if USE_SUNAUDIO
 SUNAUDIO_DIR=sunaudio
 else
@@ -77,7 +83,7 @@ else
 XIMAGE_DIR=
 endif
 
-SUBDIRS=$(DIRECTSOUND_DIR) $(OSS_DIR) $(OSS4_DIR) $(OSX_AUDIO_DIR) $(OSX_VIDEO_DIR) $(SUNAUDIO_DIR) $(V4L2_DIR) $(XIMAGE_DIR)
+SUBDIRS=$(DIRECTSOUND_DIR) $(WAVEFORM_DIR) $(OSS_DIR) $(OSS4_DIR) $(OSX_AUDIO_DIR) $(OSX_VIDEO_DIR) $(SUNAUDIO_DIR) $(V4L2_DIR) $(XIMAGE_DIR)
 
 DIST_SUBDIRS=directsound oss oss4 osxaudio osxvideo sunaudio v4l2 waveform ximage
 
--- origsrc/gst-plugins-good-0.10.30/sys/waveform/Makefile.am	2011-06-13 18:24:13.000000000 -0500
+++ src/gst-plugins-good-0.10.30/sys/waveform/Makefile.am	2011-12-18 23:04:38.572988800 -0600
@@ -1,19 +1,15 @@
 plugin_LTLIBRARIES = libgstwaveformsink.la
 
-# FIXME: Replace DIRECTSOUND CFLAGS+LIBS with waveform related ones and fix
-# the configure.ac + sys/Makefile.am to get this stuff building in MingW
-# For now, it's just disted for use in the VS builds.
-
 libgstwaveformsink_la_SOURCES =  gstwaveformsink.c gstwaveformplugin.c
 libgstwaveformsink_la_CFLAGS = $(GST_CFLAGS) $(GST_BASE_CFLAGS) \
-        $(GST_PLUGINS_BASE_CFLAGS) $(DIRECTSOUND_CFLAGS)
+        $(GST_PLUGINS_BASE_CFLAGS)
 libgstwaveformsink_la_LIBADD = \
 	$(GST_PLUGINS_BASE_LIBS) \
         -lgstaudio-$(GST_MAJORMINOR) -lgstinterfaces-$(GST_MAJORMINOR) \
         $(GST_BASE_LIBS) \
 	$(GST_LIBS) \
-	$(DIRECTSOUND_LIBS)
-libgstwaveformsink_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) $(DIRECTSOUND_LDFLAGS)
+	-lwinmm
+libgstwaveformsink_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
 libgstwaveformsink_la_LIBTOOLFLAGS = --tag=disable-static
 
 noinst_HEADERS = gstwaveformsink.h
